smqtk_relevancy/impls/rank_relevancy/sorted_sampling.py

```diff
@@ -0,0 +1,64 @@
+from typing import Hashable, Sequence, Tuple, Dict, Any, TypeVar, Type
+
+from numpy import ndarray
+
+from smqtk_core.configuration import (
+    from_config_dict,
+    make_default_config,
+    to_config_dict,
+)
+from smqtk_core.dict import merge_dict
+from smqtk_relevancy.interfaces.rank_relevancy import RankRelevancy, RankRelevancyWithFeedback
+
+T = TypeVar("T", bound="RankRelevancyWithSortedFeedback")
+
+
+class RankRelevancyWithSortedFeedback(RankRelevancyWithFeedback):
+    """
+    Wrap an instance of :class:`RankRelevancy` to provide feedback via
+    a sorted list of results
+
+    :param rank_relevancy: :class:`RankRelevancy` to use for computing
+        relevancy scores
+    :param n: Maximum number of items to return for feedback
+
+    :raises ValueError: n is negative
+
+    """
+
+    def __init__(self, rank_relevancy: RankRelevancy,
+                 n: int):
+        self._rank_relevancy = rank_relevancy
+        if n < 0:
+            raise ValueError(f"n must be nonnegative but got {n}")
+        self._n = n
+
+    def _rank_with_feedback(
+            self,
+            pos: Sequence[ndarray],
+            neg: Sequence[ndarray],
+            pool: Sequence[ndarray],
+            pool_uids: Sequence[Hashable],
+    ) -> Tuple[Sequence[float], Sequence[Hashable]]:
+        scores = self._rank_relevancy.rank(pos, neg, pool)
+        ranked = sorted(zip(scores, pool_uids), reverse=True)
```

Comment:
This will work, but a more efficient alternative would be to use `numpy.argsort`, which will return an array of indices, which can then be used to index into pool id list (assuming parallel association between pool, pool_ids, and scores, as we currently do here anyway):
```
np.asarray(pool_uids)[np.argsort(scores)]
```

Reply:
Yup, you're correct on catching the order reversal. :+1: